### ISSUE TITLE
False positives on export (destructuring) missing key object with default value

### ISSUE THREAD

**Issue by ardok** [benmosher/eslint-plugin-import#918]
```js
// constants.js
export const {
  a: {
    // `b` would have been undefined, but we will default the value with a string
    b = 'default value',
  },
} = {
  a: {
    // `b` doesn't exist here
    c: 'test',
  },
};

...

// foo.js
import {b} from './constants';

// And it results in
 error  b not found in './constants'  import/named
```

It shouldn't throw an error because we have a default value for the key there.

**Comment by ljharb** [comment]
I'm not actually sure how that named export works with destructuring - that seems like something you shouldn't be able to do.

is that exporting just a "b"?

**Comment by ardok** [comment]
It's just exporting `b`.

Code and my app run just fine because it's valid destructuring (with export).
It's just lint complaining about that.

**Comment by ljharb** [comment]
This appears to be fixed already. Happy to reopen if the current and also _next_ release doesn't solve it.
